### DISCUSSION TITLE
How to use CMake kits with a code-workspace

### DISCUSSION THREAD

**Discussion by chrisjaquet** [microsoft/vscode-cmake-tools#3943] [Q&A] [answered]
Referring to the docs [here](https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/kits.md) I am able to define CMake kits for the user (`cmake-tool-kits.json`) or for a folder (`.vscode/cmake-kits.json`) but I am having trouble figuring out how to define kits for a VSCode `.code-workspace`. I feel I am missing something obvious here.

Some context: My `.code-workspace` is set include a few projects from a larger selection within a particular folder. I am unable to use the user kits since there is a `${workspaceFolder}` substitution being done in there to find a toolchain file that is project specific, e.g. 
```
"toolchainFile": "${workspaceFolder}/BuildSystem/Toolchains/Windows.cmake",
```

The `BuildSystem` folder is added as one of the folders within the `.code-workspace` but `${workspace}` does not resolve (which I think is expected behaviour). I have a temporary workaround by adding another user kit which references `${workspace:BuildSystem}/Toolchains/Windows.cmake` but ideally I would like to specify the kit arguments in the `.code-workspace` file itself so that I can commit to VCS.

Am I missing something simple? Any advice?

**Comment by gcampbell-msft** [comment, marked as answer]
I'll start with suggesting that if you migrate to CMakePresets.json, you'd be able to have more control over it because it is based on the presence of the file in the folder.

For kits, to my knowledge, in the `.code-workspace` file, you can specify which folders are opened when you open the workspace file. In that case, you should be able to use the folder `.vscode/cmake-kits.json`. As for specifying a kit (or even a preset for that matter) for an entire workspace is not explicitly supported. 

I think the only other possible option is to define various cmake settings in the code-workspace file to help with this (for example, you could set `cmake.sourceDirectory` to the folders that are opened with the code-workspace), but we don't have any explicit support.
